### ISSUE TITLE
Mark onChain payments as "paid" (only in the frontend) once in mempool for Satspay plugin

### ISSUE THREAD

**Issue by LightRider5** [lnbits/satspay#13]
This feature request is pertinent only for onChain payments when using the Satspay plugin.

In a manner similar to BTCPayserver, it would be beneficial to allow merchants to mark a payment as "paid" in the frontend as soon as the transaction enters the mempool. This enhancement would improve user experience for onChain payments. As a safeguard, merchants could still perform an additional payment verification in the backend before confirming and processing the order.

This feature should only be optional. However, it is best practice among merchants to handle onChain payments in this manner.

**Comment by talvasconcelos** [comment]
Just to clarify when you mean optional, is that the merchant would flag if he wants the "pseudo" zero-conf behaviour or not? have a toggle, for example? and an explanation that this will only affect UI on the client side and that the charge would only be marked paid after 1 confirmation?

IMHO, it may be confusing, making it optional. Would it be better if it's "zero-conf" (for the end user UI) by default?

**Comment by talvasconcelos** [comment]
Ok, "zero conf" is working!

Now, another issue... when there's a webhook, it gets sent out when the charge is paid, it also has a button, for the user/customer to confirm the order. As per @jr21M issue ( #12  ) , which I'd like to include in this PR, how is the flow supposed to happen?

Show the button, even though the charge is not in fact confirmed (1 confirmation)?
Show only after the confirmation, meaning the user would have to wait or come back to the payment page when the payment is confirmed?
Other scenario?

CC @arcbtc

**Comment by lnbright** [comment]
tnx, one suggestion if I may:
awesome experience would be when the green tick appears quick and that will by itself trigger the order confirmation (let's say to the merchant) w/o the customer need to do anything. On card payments normally customer waits to be sent back to the merchant page (hence my re-direct suggestion above), with a banner saying ~"do not leave this page". 
But customers cannot be trusted not to hit back on the browser or put the browser in the background in the app and forget about it.

**Comment by LightRider5** [comment]
> Just to clarify, when you say "optional," do you mean the merchant can choose whether or not they want the "pseudo" zero-conf behavior? Would they, for instance, have a toggle for this? And to be clear, this would only affect the UI on the client side, and the charge would only be marked as paid after one confirmation, correct?

Yes, that seems like a sensible approach.

> In my opinion, making it optional might be confusing. Wouldn't it be clearer for the end user's UI if "zero-conf" were the default setting?

That's an excellent point. However, enabling "zero-conf" by default might lead to more confusion. Inexperienced Lnbits operators might also be more susceptible to deception by malicious customers with this setup. The approach taken by BTCPayserver, where it's optional, seems to be a good one in my view.

> Great, "zero conf" is up and running!

Impressive, that was incredibly fast!

> Now, regarding another issue: when a webhook is triggered, it's dispatched once the charge is paid. This webhook also includes a button allowing the user/customer to confirm the order. Considering @jr21M's issue (https://github.com/lnbits/satspay/issues/12), which I plan to incorporate into this PR, what should the flow look like?

My suggestion is to display the confirmation button as soon as the order is marked as "Paid" (be it an LN invoice that's settled or a pending transaction on-chain).

It might also be worth considering a "pending" flag for the webhook, especially in relation to on-chain payments.

**Comment by talvasconcelos** [comment]
I need to make it optional still... 

About your last point, this needs to be backwards compatible (i think), as so, if some merchant has an endpoint that gets hit by the webhook to trigger something, for example a download, or a shipping, whatever, they could get in trouble if not aware of this pending flag! the webhook is only triggered when the payment is confirmed! 

Guess it can be avoided with being optional and "zero-conf" is opt-in instead of default, like you suggested! It needs more in depth changes. I'll get it done soon.

**Comment by LightRider5** [comment]
It makes sense to keep everything backwards compatible. However:

> the webhook is only triggered when the payment is confirmed!

I understand. However, this poses a problem in my view. I use Lnbits solely as my backend Lightning infrastructure and handle all the frontend tasks independently. Hence, I rely exclusively on the API. If the webhook is triggered only when the payment is confirmed, then the new optional feature won't benefit my application. I believe many others primarily utilize the Lnbits API for their applications.
